crates/bvs-vault-cw20-tokenized/tests/integration_test.rs

```diff
@@ -0,0 +1,1388 @@
+            .wrap()
+            .query_wasm_smart::<cw20::AllowanceResponse>(vault.addr(), &query)
+            .unwrap();
+        assert_eq!(allowance.allowance, Uint128::new(100));
+    }
+
+    // We can decrease allowance like normal cw20 token would
+    {
+        let inner_msg = cw20_base::msg::ExecuteMsg::DecreaseAllowance {
+            spender: app.api().addr_make("spender").to_string(),
+            amount: Uint128::new(50),
+            expires: None,
+        };
+        let msg = WasmMsg::Execute {
+            contract_addr: vault.addr().to_string(),
+            msg: to_json_binary(&inner_msg).unwrap(),
+            funds: vec![],
+        };
+        let res = app.execute(staker.clone(), cosmwasm_std::CosmosMsg::Wasm(msg));
+        // let res = vault.execute(app, &staker, &inner_msg.into());
+        assert!(res.is_ok());
+
+        let query = cw20_base::msg::QueryMsg::Allowance {
+            owner: staker.to_string(),
+            spender: app.api().addr_make("spender").to_string(),
+        };
+        let allowance: cw20::AllowanceResponse = app
+            .wrap()
+            .query_wasm_smart::<cw20::AllowanceResponse>(vault.addr(), &query)
+            .unwrap();
+        assert_eq!(allowance.allowance, Uint128::new(50));
+    }
+
+    // We can transfer like normal cw20 token would
+    {
+        let inner_msg = cw20_base::msg::ExecuteMsg::Transfer {
+            recipient: app.api().addr_make("recipient").to_string(),
+            amount: Uint128::new(50),
+        };
+        let msg = WasmMsg::Execute {
+            contract_addr: vault.addr().to_string(),
+            msg: to_json_binary(&inner_msg).unwrap(),
+            funds: vec![],
+        };
+        let res = app.execute(staker.clone(), cosmwasm_std::CosmosMsg::Wasm(msg));
+        assert!(res.is_ok());
+
+        let query = cw20_base::msg::QueryMsg::Balance {
+            address: app.api().addr_make("recipient").to_string(),
+        };
+        let balance: cw20::BalanceResponse = app
+            .wrap()
+            .query_wasm_smart::<cw20::BalanceResponse>(vault.addr(), &query)
+            .unwrap();
+        assert_eq!(balance.balance, Uint128::new(50));
+    }
+
+    // We can transfer_from like normal cw20 token would
+    {
+        let inner_msg = cw20_base::msg::ExecuteMsg::TransferFrom {
+            owner: staker.to_string(),
+            recipient: app.api().addr_make("random").to_string(),
+            amount: Uint128::new(25),
+        };
+        let msg = WasmMsg::Execute {
+            contract_addr: vault.addr().to_string(),
+            msg: to_json_binary(&inner_msg).unwrap(),
+            funds: vec![],
+        };
+
+        // spender is the caller because we are using transfer_from
+        // remember spender has been given allowance of 100 and then reduced to 50
+        // in the earlier tests
+        let res = app.execute(
+            app.api().addr_make("spender"),
+            cosmwasm_std::CosmosMsg::Wasm(msg),
```

Comment:
```suggestion
msg.into()
```

More of a FYI, you can use into here.